replica/database.cc

```diff
@@ -1031,10 +1037,29 @@ bool database::update_column_family(schema_ptr new_schema) {
     return columns_changed;
 }
 
+void database::clean_disposed_column_families() {
+    dblog.info("clean_disposed_column_families");
+    for (auto it = _disposed_column_families.cbegin(); it != _disposed_column_families.cend(); /* no increment */ ) {
+        dblog.info("clean_disposed_column_families: {}", it->first );
+        if (it->second.use_count() == 1) {
+            dblog.info("clean_disposed_column_families: removing {}", it->first );
+            _disposed_column_families.erase(it++);
+        } else {
+            ++it;
+        }
+    }
+}
+
 void database::remove(const table& cf) noexcept {
     auto s = cf.schema();
     auto& ks = find_keyspace(s->ks_name());
+    auto lw_table_shared_ptr = _column_families.find(s->id());
+    _disposed_column_families[s->ks_name() + "." + s->cf_name()] = lw_table_shared_ptr->second;
```

Comment:
Maybe _disposed_column_families could contain the future object for the column family to really go away and CREATE here could co_await for that future?